Sources/Sentry/SentryClient.m

```diff
@@ -719,13 +726,35 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
         event.user.ipAddress = @"{{auto}}";
     }
 
+    BOOL eventIsATransaction
+        = event.type != nil && [event.type isEqualToString:SentryEnvelopeItemTypeTransaction];
+    BOOL eventIsATransactionClass
+        = eventIsATransaction && [event isKindOfClass:[SentryTransaction class]];
```

Comment:
m: We already check if the event is a transaction before
```suggestion
    BOOL eventIsATransactionClass
        = !eventIsNotATransaction && [event isKindOfClass:[SentryTransaction class]];
```

Reply:
I had a crash with one of the tests. We initialize an Event, and assign it the transaction type. I was not sure if this was intetnded.